### PR TITLE
Remove workaround for chef bug in git resource (fixes #289)

### LINKED ISSUE #289: Cannot force update branch error is back
### :speaking_head: Foreword

Thank for taking the time to fill this bug report fully. Without it we may not be able to fix the bug, and the issue may be closed without resolution.

## :ghost: Brief Description

Looks like the git resources changes in chef 16 are at it again. `Cannot force update the current branch.` error is back again and this time it's using the legacy `deploy` branch so i'm not sure how to fix this. I checked on the ruby-build repo and theres no branch called deploy there. Not sure if this is a cookbook or upstream issue but it would be nice to get a proper fix for this so it doesn't happen again.

## :pancakes: Cookbook version

2.4.0 (latest)

## :woman_cook:  Chef-Infra Version

```
      System Info:
      ------------
      chef_version=16.2.73
      platform=ubuntu
      platform_version=18.04
      ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
      program_name=/usr/bin/chef-client
      executable=/opt/chef-workstation/bin/chef-client
```

## :tophat: Platform details

```
      System Info:
      ------------
      chef_version=16.2.73
      platform=ubuntu
      platform_version=18.04
      ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
      program_name=/usr/bin/chef-client
      executable=/opt/chef-workstation/bin/chef-client
```

## Steps To Reproduce

add a simple ruby resource to the recipe:

```
rbenv_ruby '2.6.1' do
  user 'user'
end
```
## :police_car: Expected behavior

Successful converge

## :heavy_plus_sign: Additional context

```
System Info:
      ------------
      chef_version=16.2.73
      platform=ubuntu
      platform_version=18.04
      ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
      program_name=/usr/bin/chef-client
      executable=/opt/chef-workstation/bin/chef-client


    ================================================================================
    Error executing action `install` on resource 'rbenv_ruby[2.6.1]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    rbenv_plugin[ruby-build] (/var/chef/cache/cookbooks/ruby_rbenv/resources/ruby.rb line 41) had an error: Mixlib::ShellOut::ShellCommandFailed: git[Install ruby-build plugin] (/var/chef/cache/cookbooks/ruby_rbenv/resources/plugin.rb line 32) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '128'
    ---- Begin output of git branch -f deploy 16421762a6a7901c1c12811940187e40f7dfc3b9 ----
    STDOUT:
    STDERR: fatal: Cannot force update the current branch.
    ---- End output of git branch -f deploy 16421762a6a7901c1c12811940187e40f7dfc3b9 ----
    Ran git branch -f deploy 16421762a6a7901c1c12811940187e40f7dfc3b9 returned 128
```

### PR DESCRIPTION
**Pull request by axelrtgs**
# Description

Chef bug for git resource seems to be fixed/working now removing workaround (fixes #289) 

I think minor is appropriate given the scope of changes but we may want to bump the major to avoid startling users who may be on the chef versions affected by the bug.

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
LGTM but I wonder if this should be a patch release since it's fixing something unless this will actually change/break something.

**Comment by axelrtgs** [comment]
It changes the checkout directory and May break versions of chef that suffer from the bug

**Comment by ramereth** [comment]
@axelrtgs based on when you mentioned in #289, I think it might be a good idea to put a note in the README and the CHANGELOG letting users know that if they run into the problem, it would likely be the older versions of 16 and would need to upgrade.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ All notable changes to this project will be documented in this file.
 
 ## Unreleased
 
+**NOTE:** Some Chef version >=16 and <16.4.41 have a bug in the git resource causing some failues (see [#289](https://github.com/sous-chefs/ruby_rbenv/issues/289)). If you experience any troubles please try a more recent version of Chef 16.
```
Comment (ramereth) [inline CHANGELOG.md]:
```suggestion
**NOTE:** Some Chef versions (>= 16 and < 16.4.41) have a bug in the git resource causing some failures (see [#289](https://github.com/sous-chefs/ruby_rbenv/issues/289)). If you experience any troubles please try a more recent version of Chef 16.
```

Comment (axelrtgs) [reply CHANGELOG.md]: Fixed it in changelog and readme. thanks my engrish good today not 😛

**Review by ramereth — APPROVED** [review]

**Pull request merged**